sources/core/Stride.Core.Reflection/MemberPath.cs

```diff
@@ -357,17 +357,20 @@ public bool Apply(object rootObject, MemberPathAction actionType, object value)
         /// <param name="value">The returned value.</param>
         /// <returns><c>true</c> if evaluation of the path succeeded and the value is valid, <c>false</c> otherwise.</returns>
         /// <exception cref="System.ArgumentNullException">rootObject</exception>
-        public bool TryGetValue(object rootObject, [MaybeNullWhen(false)] out object? value)
+        public bool TryGetValue(object rootObject, out object? value)
         {
             ArgumentNullException.ThrowIfNull(rootObject);
             value = null;
             try
             {
-                object nextObject = rootObject;
+                object? nextObject = rootObject;
                 for (int i = 0; i < items.Count; i++)
                 {
                     var item = items[i];
-                    nextObject = item.GetValue(nextObject)!;
+                    if(nextObject is not null)
+                    {
+                        nextObject = item.GetValue(nextObject);
+                    }
                 }
                 value = nextObject;
             }
```

Comment:
How about:
```cs
try
{
    object nextObject = rootObject;
    for (int i = 0; i < items.Count; i++)
    {
        var item = items[i];
        var newObject = item.GetValue(nextObject);
        if (newObject is null)
        {
        	value = null;
            return true;
        }
        nextObject = newObject;
    }
    value = nextObject;
}
```

Reply:
[291b3ea](https://github.com/stride3d/stride/pull/2266/commits/291b3ea507e619f175b1913538f70aff024ad851)
did that